pandas/tests/io/test_parquet.py

```diff
@@ -212,28 +212,39 @@ def check_error_on_write(self, df, engine, exc):
             with tm.ensure_clean() as path:
                 to_parquet(df, path, engine, compression=None)
 
-    def check_round_trip(self, df, engine, expected=None,
-                         write_kwargs=None, read_kwargs=None,
-                         check_names=True):
```

Comment:
pls revert this, it is easy enough simply to passing ``path=None`` as the default (and if its None use ``ensure_clean`` to make sure things are cleaned), for the s3 you can pass in the path.

It is VERY confusing to have 2 check functions, and so now future readers have no idea what to use. This is just increasing complexity.
Future you are changing the existing naming scheme by changing engine -> engine_impl which is non-standard.

Reply:
the change in the write_kwargs is ok.

Reply:
This makes `check_round_trip` be very not DRY. 

```python
    def check_round_trip(self, df, engine, path=None, expected=None,
                         write_kwargs=None, read_kwargs=None,
                         check_names=True, ):

        if write_kwargs is None:
            write_kwargs = {'compression': None}

        if read_kwargs is None:
            read_kwargs = {}

        if path is None:
            with tm.ensure_clean() as path:
                df.to_parquet(path, engine, **write_kwargs)
                actual = read_parquet(path, engine, **read_kwargs)

                if expected is None:
                    expected = df

                tm.assert_frame_equal(expected, actual, check_names=check_names)

                # repeat
                df.to_parquet(path, engine, **write_kwargs)
                actual = read_parquet(path, engine, **read_kwargs)

                if expected is None:
                    expected = df

                tm.assert_frame_equal(expected, actual, check_names=check_names)
        else:
            df.to_parquet(path, engine, **write_kwargs)
            actual = read_parquet(path, engine, **read_kwargs)

            if expected is None:
                expected = df

            tm.assert_frame_equal(expected, actual, check_names=check_names)

            # repeat
            df.to_parquet(path, engine, **write_kwargs)
            actual = read_parquet(path, engine, **read_kwargs)

            if expected is None:
                expected = df

            tm.assert_frame_equal(expected, actual, check_names=check_names)
```

Any suggestions?